urllib3/response.py

```diff
@@ -351,3 +361,71 @@ def readinto(self, b):
         else:
             b[:len(temp)] = temp
             return len(temp)
+
+    def read_chunked(self, amt=None):
```

Comment:
This method is pretty meaty, would be a good candidate for somekind of helper. Maybe a ChunkedReader class?

Not required for this PR, just something to think about. :)